File: kickstart_coding_practice_2_2022/building_palindromes/starter_code.cc
#include <bits/stdc++.h>

using namespace std;

int positiveAnswersCount(int N, int Q, string blockCharacters, vector<int> L,
                         vector<int> R) {
  // TODO: Complete this function to count number of questions Anna can answer
  // "yes" to

  
  return 0;
}

int main() {
  int T;
  // Get number of test cases
  cin >> T;
  for (int tc = 1; tc <= T; tc++) {
    int N, Q;
    // Get number of blocks and number of questions
    cin >> N >> Q;

    string blockCharacters;
    // Get the string representing the characters on the blocks
    cin >> blockCharacters;

    vector<int> L(Q), R(Q);
    for (int i = 0; i < Q; i++) {
      cin >> L[i] >> R[i];
    }

    cout << "Case #" << tc << ": "
         << positiveAnswersCount(N, Q, blockCharacters, L, R) << endl;
  }
  return 0;
}

File: kickstart_coding_practice_2_2022/building_palindromes/starter_code.java
import java.util.*;

public class Solution {

  static boolean canFormPalindrome(char[] blocks, int left, int right) {
    boolean isPalindrome = false;

    // TODO: implement logic to determine whether blocks[l]..blocks[r] can form a palindrome
    // NOTE: remember that blocks[] indexes start at 1, so blocks[0] is empty!

    return isPalindrome;
  }

  public static void main(String[] args) {
    Scanner scanner = new Scanner(System.in);
    // Get number of test cases in input
    int testCaseCount = scanner.nextInt();
    // Iterate through test cases
    for (int tc = 1; tc <= testCaseCount; ++tc) {
      // Get number of blocks & questions for this test case
      int numberOfBlocks = scanner.nextInt();
      int numberOfQuestions = scanner.nextInt();

      // Read consecutive block characters as a single string
      String blockString = scanner.next();
      // Create array of characters for each block, starting with index 1
      char[] blocks = new char[numberOfBlocks + 1];
      for (int i = 0; i < numberOfBlocks; ++i) {
        blocks[i + 1] = blockString.charAt(i);
      }
      int palindromeCount = 0;
      // Iterate through questions for this test case
      for (int i = 0; i < numberOfQuestions; ++i) {
        // Get left-index and right-index (inclusive) for this question
        int left = scanner.nextInt();
        int right = scanner.nextInt();
        // Create a copy of blocks[] array so that canFormPalindrome()
        // can safely modify it without affecting other questions
        char[] blockCopy = Arrays.copyOf(blocks, blocks.length);
        // Increment count if this question's sub-array can form a palindrome
        if (canFormPalindrome(blockCopy, left, right)) {
          ++palindromeCount;
        }
      }
      // Print results
      System.out.println("Case #" + tc + ": " + palindromeCount);
    }
  }
}
